Homework11/WebTableTests.cs

```diff
@@ -0,0 +1,141 @@
+            var submitButton = _driver.FindElement(By.XPath("//button[@id='submit']"));
+            //fill in form inputs
+            FillInInput(fistNameInput, Constants.firstName);
+            FillInInput(lastNameInput, Constants.lastName);
+            FillInInput(emailInput, Constants.email);
+            FillInInput(ageInput, Constants.age);
+            FillInInput(salaryInput, Constants.salary);
+            FillInInput(departmentInput, Constants.department);
+            submitButton.Click();
+            //get text of new entry
+            ReadOnlyCollection<IWebElement> gridcell = _driver.FindElements(By.XPath("((//div[@role='row'])[5])/div"));
+            Assert.AreEqual(ExpectedConstants.newEntryValue, GetTextOfCollectionElements(gridcell));
+        }
+
+        [Test]
+        public void EditFirstEntry()
+        {
+            var editButton = _driver.FindElement(By.XPath("//span[@title='Edit']"));
+            editButton.Click();
+            //find form inputs
+            var fistNameInput = _driver.FindElement(By.XPath("//input[@id='firstName']"));
+            var lastNameInput = _driver.FindElement(By.XPath("//input[@id='lastName']"));
+            var emailInput = _driver.FindElement(By.XPath("//input[@id='userEmail']"));
+            var ageInput = _driver.FindElement(By.XPath("//input[@id='age']"));
+            var salaryInput = _driver.FindElement(By.XPath("//input[@id='salary']"));
+            var departmentInput = _driver.FindElement(By.XPath("//input[@id='department']"));
+            var submitButton = _driver.FindElement(By.XPath("//button[@id='submit']"));
+            //clear form inputs
+            DeleteValue(fistNameInput);
+            DeleteValue(lastNameInput);
+            DeleteValue(emailInput);
+            DeleteValue(ageInput);
+            DeleteValue(salaryInput);
+            DeleteValue(departmentInput);
+            //fill in form inputs
+            FillInInput(fistNameInput, Constants.firstName);
+            FillInInput(lastNameInput, Constants.lastName);
+            FillInInput(emailInput, Constants.email);
+            FillInInput(ageInput, Constants.age);
+            FillInInput(salaryInput, Constants.salary);
+            FillInInput(departmentInput, Constants.department);
+            submitButton.Click();
+            //get text of updated entry
+            ReadOnlyCollection<IWebElement> gridcell = _driver.FindElements(By.XPath("((//div[@role='row'])[2])/div"));
+            Assert.AreEqual(ExpectedConstants.newEntryValue, GetTextOfCollectionElements(gridcell));
+        }
+
+        [Test]
+        public void DeleteFirstEntry()
+        {
+            var deleteButton = _driver.FindElement(By.XPath("//span[@title='Delete']"));
+            deleteButton.Click();
+            var expectedCell = _driver.FindElements(By.XPath("//div[contains(text(),'Vega')]"));
+            Assert.AreEqual(0, expectedCell.Count);
```

Comment:
Yes it works, but it can fail with other data. I would do next steps: remember current list of elements or the first elemt (e.g. First Name), delete first element, get new results and compare

Reply:
fixed